### DISCUSSION TITLE
assets not working in production mode

### DISCUSSION THREAD

**Discussion by vazand** [playframework/playframework#11392] [Q&A] [answered]
While running my play application in DEV mode,Users can upload images and  uploaded images shown in web, but in PRODUCTION mode users can upload images but their images not shown in the web. 
  twirl  template code ->          `<img src="@routes.Assets.versioned(listOfFiles(1))" width=25% height=auto alt="image1.png">`
(the listOfFiles(1) is an image location -> (public/image1.png)) 
Could you help me for a second?...!

**Comment by Squeng** [comment, marked as answer]
I depends on how you build/deploy/run your app. (For example, I use sbt stage: https://github.com/Squoss/Squawg/blob/main/Dockerfile)

For example, in DEV the /public/images as explained in https://www.playframework.com/documentation/2.8.x/Anatomy is the only instance. In PRODUCTION there might be several in your case, one below /public because you still upload to there and one below /target. I would look into that if you haven't done so already.

And as a general rule, I would use a third-party storage for user uploads if that's possible in your case, https://www.clever-cloud.com/cellar-s3-hosting/ for instance.

I hope this helps!

**Reply by vazand** [reply]
@Squeng Thanks,
After reading some docs ,I've found  [https://www.playframework.com/documentation/2.8.x/ScalaStream](url) 
`Ok.sendFile(new java.io.File("/path/to/file"))` is used to serve a file.

**Reply by Squeng** [reply]
And just in case, there's also sendReference: https://github.com/Squoss/Squawg/blob/main/beapi/app/gui/ReactController.scala#L57
